alacritty/src/display/mod.rs

```diff
@@ -188,9 +188,17 @@ pub struct Display {
 
 impl Display {
     pub fn new<E>(config: &Config, event_loop: &EventLoop<E>) -> Result<Display, Error> {
-        // Guess DPR based on first monitor.
-        let estimated_dpr =
-            event_loop.available_monitors().next().map(|m| m.scale_factor()).unwrap_or(1.);
+        #[cfg(any(not(feature = "x11"), target_os = "macos", windows))]
+        let is_x11 = false;
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        let is_x11 = event_loop.is_x11();
+
+        // Guess DPR based on first monitor, except on Wayland.
+        let estimated_dpr = if cfg!(macos) || cfg!(windows) || is_x11 {
```

Comment:
I think we should check `target_os = "macos"`, rather than just macos, since it's technically unix and not macos so the target_os is necessary iirc.

Reply:
Ohhhh, glad I finally had to learn about this. The `target_family` configuration options are also re-exported as options themselves. I've been confused about that for a while TBH. Turns out `=` binds tighter than `,` in `cfg` after all, I was just misunderstanding the whole thing!